lib/rspec-puppet-facts.rb

```diff
@@ -348,4 +343,35 @@ def self.down_facter_version(version, minor_subtractor = 1)
     minor = (minor.to_i - minor_subtractor).to_s
     "#{major}.#{minor}.#{z}"
   end
+
+  def self.facter_version_for_puppet_version(puppet_version)
+    return Facter.version if puppet_version.nil?
+    json_path = File.expand_path(File.join(__dir__, '..', 'ext', 'puppet_agent_components.json'))
+    return Facter.version unless File.file?(json_path) && File.readable?(json_path)
+
+    fd = File.open(json_path, 'rb')
+    data = JSON.parse(fd.read)
```

Comment:
Is it worth rescuing a JSON parse error here?

Reply:
:+1: